### ISSUE TITLE
firefox 90 hangs when handling alert

### ISSUE THREAD

**Issue by derah4gh** [mozilla/geckodriver#1913]
## System 1 Linux

* Geckodriver Version: 0.22.0
* Platform: Linux Debian 4.9
* Firefox: 90.x
* robotframework: 3.1.2
* robotframework-seleniumlibrary: 4.3.0
* selenium: 3.141.0
 
## System 2 Windows 10

* Geckodriver Version: 0.29.1
* Platform: Windows 10 64bit
* Firefox: 90.0.2 64 bit
* robotframework: 3.2.2
* robotframework-seleniumlibrary: 5.1.3
* selenium: 3.141.0

## Webpage

Note: This is a simplified version found on several locations in our App
This code is "check a form with JavaScript before submit"

```
<html>
 <head>
  ...
  <script type="text/javascript"> 
<!--
function check_form(){
  if( form_is_ok() ) {  
    document.myform.submit()
  }else{
    alert("Please fill all values!");
  }
}
-->
</script>
</head>
<body>
  ...
  <form name="myform" action="...">  
    ...
    <p><a href="javascript:check_form()">Submit Form</a></p>
    ...
  </form>
</body>
</html>
````

## Robot Testcase

````
Form Submit fails
    ...
    # do not fill any fields
   Click Link     Submit Form
   Alert Should be Present     Please fill all values!
   ...
````
Note:  The Problem is located in Robot's "Handle Alert",  which is used by "Alert Should be Present"

## Expected 

* Alert comes up
* Test checks content of alert --> ok
* Test accepts alert
* Alert disappears
* Test continues

## Got

Always: The alert disappears and then the test hangs, firefox window remains open.

## Important Versions Information

* This issue came up with Firefox 90: After auto-update on some of our test mashines several tests always failed there, version before 90 worked fine
* The issue is on Windows and on Linux
* Browsers Firefox < 90, Edge and Chrome (Windows) work without problem

## Trace-level log

[geckodriver-1.log](https://github.com/mozilla/geckodriver/files/6906919/geckodriver-1.log)

**Comment by whimboo** [comment]
@derah4gh is the handling of the form/alert done in a popup window or a regular browser window/tab? I can see that a new window gets opened during the test. If it's a popup please test with Firefox 91 beta, which recently got a [fix for popup windows](https://bugzilla.mozilla.org/show_bug.cgi?id=1721982).

**Comment by derah4gh** [comment]
The handling is done in a popup window. With Firefox 91.0b9 64bit on Windows 10 our tests now are working again. The "fix for popup windows" you mentioned also fixed this problem. Many thanks for your support.

**Comment by whimboo** [comment]
That's great to hear! We will ship the fix soon with the 0.30.0 release.
